src/scenes/ui/BattleUI.cs

```diff
@@ -0,0 +1,25 @@
+using Godot;
+using System;
+
+public class BattleUI : CanvasLayer
+{
+    private String currentTower;
+
+    public override void _Ready()
+    {
+        
+    }
+
+    public void onTowerSelected(TowerButton towerButton) {
+        if (towerButton != null) {
+            currentTower = towerButton.TowerType;
+        }
+        foreach (Node node in GetChildren())
+        {
+            if (node is TowerButton) {
+                TowerButton tower = (TowerButton)node;
+                tower.Active = tower == towerButton ? true : false;
```

Comment:
The `? true : false` part is not necrssary. Consider to remove it

Reply:
oh my god